Manage dirty flags on all inputs

hideheader opened the pull request:
1. Record initial value of text, select, and checkbox elements and re/set a dirty class on each when the value changes.
2. Combine the individual flags into a global dirty flag.
3. Add 'dirty' status to the document title.

JasonBarnabe commented:
I can barely understand the whole `clean` function. Since I'm the one who has to support this going forward, I'm not going to include it as is. Specifically:

- Don't overload the function - have separate `isClean` and `setClean` functions.
- Drop the anonymous function.
- Since the stuff on this page is the only thing that's ever going to call these functions, you don't need to type check parameters everywhere.

hideheader commented:
> I can barely understand the whole clean function.

Oh, pshaw.

JasonBarnabe commented:
Thanks.

I'm unsure if this was an oversight or something you intentionally didn't do, but removing a section or "applies to" doesn't seem be noticed as a change.

I'm also seeing what may be a Chrome bug - the second time you trigger the onbeforeunload, the tab loses its title.

hideheader commented:
> an oversight or something you intentionally didn't do

Yes. Intentional (adding then removing a section or a filter shouldn't sully the document), but an oversight (I didn't consider this case). Fixed.

`body` is now `.dirty` when anything else is `.dirty`. This may be helpful:
````css
.dirty {outline: 2px solid red;}
body.dirty {outline: none}
body.dirty::before {
  content: "dirty"; 
  position: fixed; top: 0mm; left: 1mm;
  font-size: 6mm; color: red; 
}
````

> the second time you trigger the onbeforeunload, the tab loses its title.

Can't reproduce it. (1) Edit style, (2) append " grr" to Name, (3) click Back to Manage, (4) repeat (3), (5) no joy. Same result with (3') call `onbeforeunload()` in the Console.

tophf commented on edit.js:
```diff
@@ -12,10 +14,73 @@ appliesToEverythingTemplate.innerHTML = t("appliesToEverything") + ' <button cla
+	items[id] = clean;
+
+	if (clean) node.classList.remove("dirty");
+	else node.classList.add("dirty");
```
Until recently there were only 8 braceless ifs in Stylish ([2 in install.js](https://github.com/JasonBarnabe/stylish-chrome/blob/3fddfae15a7e34d2f8c32c0d58a5e644a4dc97e4/install.js#L40-L43) and [6 in manage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/b019038f5701b67d475a36275fcbe69ac5d6ee04/manage.js#L59-L72)), then @hideheader added another 8 or more if the `else` parts are also counted ([2 in edit.js](https://github.com/JasonBarnabe/stylish-chrome/blob/284d4b8ec89babc088b2b8b6eed839ec297b0ec9/edit.js#L80-L83), [1 in popup.js](https://github.com/JasonBarnabe/stylish-chrome/blob/f8600c2fb3b377dc185d07533be0b80da4865a0b/popup.js#L57), [4 in storage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/e7f6c62e0178905233c0c420cc8de758a22fd3af/storage.js#L196-L215), [1 in manage.js](https://github.com/JasonBarnabe/stylish-chrome/blob/a0482d3f0b903adefbc2e092741d6ea4d5dcade0/manage.js#L357)).

For personal _write-only_ small stuff I use the compact one-line braceless form. Some large projects also use it (CodeMirror for example). However generally it's considered bad practice because it may lead to errors when another statement is added to the braceless block without being cautious enough to add the braces.

So, @hideheader, @JasonBarnabe, should we enforce braces on `if-else` or not? Should all one-line `if() action;` be expanded to 3 lines?

JasonBarnabe replied:
I prefer braces and multi-line `if`s, but I wouldn't reject a PR if it had code in another format.

tophf commented:
There is a problem: when I edit an applies-to url the overall `*` indicator in the tab title doesn't appear. However when I close the tab a warning messagebox is shown. If I cancel it, the indicator appears.

hideheader commented:
Yes. [[1]](https://github.com/JasonBarnabe/stylish-chrome/pull/70#issuecomment-81213459) [[2]](https://github.com/JasonBarnabe/stylish-chrome/pull/70/files#diff-3cc18e14de6e30bc468459dbeb194f7dR19)
